web/modules/custom/solution/src/SolutionAffiliationFieldItemList.php

```diff
@@ -102,55 +127,67 @@ protected function getAffiliation(): array {
    * collection cache in order to reflect the changes on Drupal API level.
    */
   protected function updateAffiliation(): void {
-    /** @var \Drupal\sparql_entity_storage\Database\Driver\sparql\ConnectionInterface $sparql */
-    $sparql = \Drupal::service('sparql.endpoint');
     /** @var \Drupal\rdf_entity\RdfInterface $solution */
     $solution = $this->getEntity();
-    /** @var \Drupal\sparql_entity_storage\SparqlEntityStorageGraphHandlerInterface $graph_handler */
-    $graph_handler = \Drupal::service('sparql.graph_handler');
-    $graph_uri = $graph_handler->getBundleGraphUri('rdf_entity', 'collection', $solution->get('graph')->target_id);
+    $connection = $this->getSparqlConnection();
+    $graph_uris = $this->getAvailableGraphs();
+
     /** @var \Drupal\sparql_entity_storage\SparqlEntityStorageFieldHandlerInterface $field_handler */
     $field_handler = \Drupal::service('sparql.field_handler');
     $field_uri = $field_handler->getFieldPredicates('rdf_entity', 'field_ar_affiliates')['collection'];
 
-    // Get existing affiliation.
-    $select[] = 'SELECT ?id';
-    $select[] = "FROM <{$graph_uri}>";
-    $select[] = "WHERE { ?id <{$field_uri}> <{$solution->id()}> . }";
-    $select[] = "ORDER BY (?id)";
-    $existing_ids = [];
-    foreach ($sparql->query(implode("\n", $select)) as $item) {
-      $existing_ids[] = (string) $item->id;
-    }
+    $affected_ids = [];
+    foreach ($graph_uris as $graph_uri) {
+      // Get existing affiliation.
+      $select = [];
+      $select[] = 'SELECT ?id';
+      $select[] = "FROM <{$graph_uri}>";
+      $select[] = 'WHERE {';
+      $select[] = "?id <{$field_uri}> <{$solution->id()}> .";
+      // Ensure the entity exists in the graph.
+      $select[] = "?id a ?type .";
+      $select[] = '}';
+      $select[] = "ORDER BY (?id)";
+
+      $existing_ids = [];
+      foreach ($connection->query(implode("\n", $select)) as $item) {
+        $existing_ids[] = (string) $item->id;
+      }
 
-    $new_ids = array_map(function (EntityReferenceItem $field_item): string {
-      return $field_item->target_id;
-    }, $this->list);
-    sort($new_ids);
+      $new_ids = array_map(function (EntityReferenceItem $field_item): string {
+        return $field_item->target_id;
+      }, $this->list);
+      sort($new_ids);
 
-    // The affiliation has been preserved. Exit here.
-    if ($new_ids === $existing_ids) {
-      return;
-    }
+      // The affiliation has been preserved. Exit here.
+      if ($new_ids === $existing_ids) {
+        return;
```

Comment:
You are completely right. I think this was there from before when we were checking the specific graph and I missed this. Thanks.

Reply:
This was reverted to remain a return since the foreach loop is now removed. We do not need to iterate through all graphs and run a separate query. We can simply go with a single query for this.